### DISCUSSION TITLE
Where is TestRig?

### DISCUSSION THREAD

**Discussion by ghost** [antlr/antlr4#3908] [Q&A]
Yes I know, it's in org.antlr.v4.runtime.misc.TestRig
Here's the javadoc:  https://www.antlr.org/api/Java/org/antlr/v4/runtime/misc/TestRig.html

The description says "A proxy for the real org.antlr.v4.gui.TestRig that we moved to tool artifact from runtime."
And that's the end of the description.
But wait! It's Deprecated! In Java, that means don't use it, and they usually suggest a better way.

So, my question _really_ is:
Where is the real TestRig? Is it called something else? Does it do anything else?

**Comment by chris-miner** [comment]
If you are looking for source, TestRig.java is under `tool/src/org/antlr/v4/gui/TestRig.java`

The class is part of the antlr4-VERSION.jar (replace VERSION)  For example, antlr4-4.9.3.jar or antlr4-4.11.1.jar 

Also, in that case, it isn't deprecated.

**Reply by ghost** [reply]
Thanks Chris,
I thought about the functions that TestRig must do, and I was on the right track, but I did not know where to look for it in the source,
But seeing the source makes it all clear. A little help here goes a long way.
